app/controllers/events_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+
+# Class...
+class EventsController < ApplicationController
+  def index
+    @events = Event.all
+  end
+
+  def show
+    @event = Event.find(params[:id])
+  end
```

Comment:
You are repeating this line many times. Instead, use a before_action filter to set the event.

You can see the documentation at:
https://guides.rubyonrails.org/action_controller_overview.html#filters

Reply:
Fixed!